### PR TITLE
Switch to rclcpp logging and improve messages

### LINKED ISSUE #166: Failed to get response from ROS service runtime_error should be informative
## Feature request
runtime_error should be informative.
#### Expected behavior
`"what(): Failed to get response from ROS service: myservice/"`
#### Actual behavior
`"what(): Failed to get response from ROS service"`

#### Implementation considerations
Change the code in:
https://github.com/ros2/ros1_bridge/blob/961ddee6b76bcb670ce3cb939d125044de820102/include/ros1_bridge/factory.hpp#L252
to 
```cpp
throw std::runtime_error("Failed to get response from ROS service: " + client.getService());
```
### Optional
Maybe it should include the fact that the error was thrown from the `ros1_bridge`

### PR DESCRIPTION
**Pull request by paulbovbel**
Switch to rclcpp logging and improve messages (Close #166)

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Please consider creating two separate pull requests for these two unrelated changes. That will allow both patches to be reviewer easier and move forward independent of each other.

**Comment by paulbovbel** [comment]
Done

File: include/ros1_bridge/factory.hpp [hunk 1]
```diff
@@ -146,20 +144,23 @@ class Factory : public FactoryInterface
...
+    const std::string & ros2_type_name,
+    rclcpp::Logger logger)
   {
     typename rclcpp::Publisher<ROS2_T>::SharedPtr typed_ros2_pub;
     typed_ros2_pub =
       std::dynamic_pointer_cast<typename rclcpp::Publisher<ROS2_T>>(ros2_pub);
 
     if (!typed_ros2_pub) {
-      throw std::runtime_error("Invalid type for publisher");
+      throw std::runtime_error("Invalid type " + ros2_type_name + " for ROS2 publisher " +
```
Comment (dirk-thomas) [inline include/ros1_bridge/factory.hpp]: Please wrap after the open parenthesis and intend all following line one level / 2 spaces.

Same below.

Comment (dirk-thomas) [reply include/ros1_bridge/factory.hpp]: When mentioning ROS 1 or ROS 2 in the error message please use a space before the number.

Same below.

File: include/ros1_bridge/factory.hpp [hunk 2]
```diff
@@ -241,34 +241,37 @@ class ServiceFactory : public ServiceFactoryInterface
...
     auto client = std::dynamic_pointer_cast<rclcpp::Client<ROS2_T>>(cli);
     if (!client) {
-      fprintf(stderr, "Failed to get the client.\n");
+      RCLCPP_ERROR(logger, "Failed to get ROS2 client " + service_name);
       return false;
     }
     auto request2 = std::make_shared<ROS2Request>();
     translate_1_to_2(request1, *request2);
-    while (!client->wait_for_service(std::chrono::seconds(1))) {
+    while (!client->wait_for_service(std::chrono::seconds(5))) {
```
Comment (dirk-thomas) [inline include/ros1_bridge/factory.hpp]: This change is also unrelated to the logging. Why was the timeout changed?

File: include/ros1_bridge/factory.hpp [hunk 3]
```diff
@@ -241,34 +241,37 @@ class ServiceFactory : public ServiceFactoryInterface
   using ROS2Response = typename ROS2_T::Response;
 
   void forward_2_to_1(
-    ros::ServiceClient client, const std::shared_ptr<rmw_request_id_t>,
+    ros::ServiceClient & client, rclcpp::Logger logger, const std::shared_ptr<rmw_request_id_t>,
```
Comment (dirk-thomas) [inline include/ros1_bridge/factory.hpp]: The added reference seems to be unrelated to the logging change. Please describe why it was added.

File: include/ros1_bridge/factory.hpp (hunk 1 above)
Comment (paulbovbel) [reply include/ros1_bridge/factory.hpp]: > Same below.

Sorry, which below? I imagine not after every open parenthesis? Is there a preferred style here (albeit not one covered by the linter)?

Comment (dirk-thomas) [reply include/ros1_bridge/factory.hpp]: > I imagine not after every open parenthesis? Is there a preferred style here

Yes, whenever parameters are wrapped all of them should wrapped (or none). Not a few following the parenthesis and a few being wrapped.

File: include/ros1_bridge/factory.hpp (hunk 3 above)
Comment (paulbovbel) [reply include/ros1_bridge/factory.hpp]: Thanks, carried over from the unrelated changes, purging.

File: include/ros1_bridge/factory.hpp (hunk 2 above)
Comment (paulbovbel) [reply include/ros1_bridge/factory.hpp]: Purged, thanks

File: include/ros1_bridge/factory.hpp (hunk 1 above)
Comment (paulbovbel) [reply include/ros1_bridge/factory.hpp]: Uncrustify wants:

```
    if (!typed_ros2_pub) {
      throw std::runtime_error(
              "Invalid type " + ros2_type_name + " for ROS 2 publisher " +
              ros2_pub->get_topic_name());
    }
```

Comment (paulbovbel) [reply include/ros1_bridge/factory.hpp]: > Yes, whenever parameters are wrapped all of them should wrapped (or none).

Fixed, thanks.

**Comment by paulbovbel** [comment]
Fixed up PR with requested changes, thanks for the dilligence

**Comment by dirk-thomas** [comment]
[Build Status](https://ci.ros2.org/job/ci_packaging_linux/171/)

**Comment by dirk-thomas** [comment]
@paulbovbel Please rebase your branch on top of the latest state of the target branch.

**Comment by paulbovbel** [comment]
Done!

**Comment by dirk-thomas** [comment]
[Build Status](https://ci.ros2.org/job/ci_packaging_linux/172/)

**Comment by paulbovbel** [comment]
Fixed.

**Comment by dirk-thomas** [comment]
[Build Status](https://ci.ros2.org/job/ci_packaging_linux/173/)

**Comment by dirk-thomas** [comment]
> Fixed.

Please run the packages tests locally to avoid this extra iteration in the future.

**Comment by dirk-thomas** [comment]
Thanks for the patch.

File: include/ros1_bridge/factory.hpp [hunk 4]
```diff
@@ -146,20 +144,23 @@ class Factory : public FactoryInterface
...
+      throw std::runtime_error(
+              "Invalid type " + ros2_type_name + " for ROS 2 publisher " +
+              ros2_pub->get_topic_name());
     }
 
     const boost::shared_ptr<ros::M_string> & connection_header =
       ros1_msg_event.getConnectionHeaderPtr();
     if (!connection_header) {
-      printf("  dropping message without connection header\n");
+      RCLCPP_WARN(logger, "Dropping ROS 1 message %s without connection header", ros1_type_name);
```
Comment (dirk-thomas) [inline include/ros1_bridge/factory.hpp]: The non-format argument must not be a `std::string`. See #182 for the fix.
